release.md

```diff
@@ -0,0 +1,298 @@
+| gnosis/pm-contracts  |      34 |  21.7s |  10.9s |     10.2s |
+| zeppelin-solidity    |     107 |  36.7s |  11.7s |     11.1s |
+
+
+For help installing a natively built compiler, see the Solidity docs [here](https://solidity.readthedocs.io/en/v0.4.23/installing-solidity.html#binary-packages).
+
+### Contract Profiling
+Truffle has always tried figure out which contracts changed recently and compile the smallest set necessary (with varying degrees of success). It now does this reliably so you may benefit from setting up an `npm` convenience script that looks like this:
+```
+"test": "truffle compile && truffle test"
+```
+
+Many thanks to the Solidity team for making all of the above possible and for their helpful advice over the last year.  🙌 🙌
+
+
+# 2. 🐇 🐇  Web3 1.0  🐇 🐇
+
+`truffle-contract` now uses Web3 1.0 under the hood.  It's nice! The error handling (especially the parameter checking) is really good. And there's lots of new work happening over there  - ENS support is being added and [EthPrize](http://ethprize.io/) is funding an API to make contract packages published to EthPM easily available as well.
+
+We've tried to minimize the number of changes required to transition an existing test suite from Web3 0.x to 1.0. Unfortunately there are some breaking changes due to differences in the way Web3 1.0 formats outputs (see chart below).
+
+The biggest change is that `truffle-contract` now returns numbers as [BN](https://github.com/indutny/bn.js/) by default instead of BigNumber. This is necessary - BigNumber doesn't have the precision required by the EVM. The number libraries share some methods (notably `.toNumber()`) but their arithmetic operations are quite different.
+
+**Important**: If you receive a number directly from web3 (e.g. *not* from a truffle-contract instance)  you'll get a **string**. Luckily there's a helpful BN conversion method in the web3 utils.
+```javascript
+const stringBalance = await web3.eth.getBalance('0xabc..');
+const bnBalance = web3.utils.toBN(stringBalance);
+```
+
+**Helpful Resources**
++  Web3 1.0's documentation. Well worth taking a look - lots of differences in their new API.
++ EthWork's [bn-chai](https://github.com/EthWorks/bn-chai) library: BN helpers for your tests. (💡 ProTip courtesy elenadimitrova)
+
+------------
+### ⚠️ Breaking Changes ⚠️
+
+| Category | V4 (Web3 0.0) |  V5 (Web3 1.0) |
+| ------ | ---- | ------- |
+| addresses (return value) | lower-case | check-summed (mixed-case) |
+| numbers  (return value)  | BigNumber | BN (configurable) |
+| tuples (return value)       | Array | Object w/ named & indexed keys |
+| `.contract` (underlying Web3 abstraction)| same as now | completely [different](https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html) |
+|`.at` (TruffleContract method)| sync / then-able | async |
+
+-----------
+### 🍨 Features 🍨
+
+### Methods / `.new` have an EventEmitter interface *and*  return a promise.
+```javascript
+example
```

Comment:
For these examples, I think it's helpful to include a line above:
```javascript
const example = await artifacts.require("Example").deployed();
```

Well, maybe not, because that kind of clashes with the `.then()` syntax we're showing off.

Wait, is this now possible?:
```javascript
let receipt = await example
  .setValue(123)
  .on('transactionHash', hash => {})
  /* ... */
  .on('confirmation', (num, receipt) => {} )
```

Anyway, my point is that it's probably worth showing a more complete example so there's literally no guesswork.

Reply:
Mmm. Yeah I could go either way - I think I had it as you've shown it originally but when I looked at the web3 docs they used the `then` which seemed clearer. 

Like there's also a weird temporal thing about lines above happening before lines below. In the receipt example  - receipt resolves *after* the txHash fires and that seemed weird to me. 

Am I crazy lol?

Reply:
Added the require.